pkg/operators/ebpf/ebpf.go

```diff
@@ -368,8 +371,19 @@ func (i *ebpfInstance) Prepare(gadgetCtx operators.GadgetContext) error {
 	}
 
 	// Create network tracers, one for each socket filter program
+	// The same applies to uprobe / uretprobe as well.
 	for _, p := range i.collectionSpec.Programs {
 		switch p.Type {
+		case ebpf.Kprobe:
+			if strings.HasPrefix(p.SectionName, "uprobe/") ||
+				strings.HasPrefix(p.SectionName, "uretprobe/") {
+				uprobeTracer, err := uprobetracer.NewTracer[api.GadgetData](gadgetCtx.Logger())
+				if err != nil {
+					i.Close()
+					return fmt.Errorf("creating uprobe tracer: %w", err)
+				}
+				i.uprobeTracers[p.Name] = uprobeTracer
+			}
```

Comment:
Below on line 475, there is:
```
	if len(i.tcHandlers) > 0 {
		// For now, override enrichment
		gadgetCtx.SetVar("NeedContainerEvents", true)
```

@flyth Do we need to add `NeedContainerEvents=true` for `i.networkTracers` and `i.uprobeTracers` too?

Why are tc programs different than socket filters?

Reply:
Currently, you need that whenever the code depends on something to be called inside `AttachContainer()` and `DetachContainer()` (so per-container) and the output struct is not using  `mntnsid` or `netnsid`.